cpp/src/lance/io/writer.cc

```diff
@@ -108,7 +108,28 @@ ::arrow::Status FileWriter::WritePrimitiveArray(const std::shared_ptr<format::Fi
                                                 const std::shared_ptr<::arrow::Array>& arr) {
   auto field_id = field->id();
   auto encoder = field->GetEncoder(destination_);
-  ARROW_ASSIGN_OR_RAISE(auto pos, encoder->Write(arr));
+  auto type = field->type();
+
+  ::arrow::Result<std::shared_ptr<::arrow::Array>> physical_arr;
+  switch (type->id()) {
+    case ::arrow::TimestampType::type_id:
+    case ::arrow::Date64Type::type_id:
+    case ::arrow::Time64Type::type_id:
+      physical_arr = arr->View(::arrow::int64());
```

Comment:
nit: is physical_arr just storage_arr? or is the concept here different?

Reply:
Yes, they are the same. I can change the name to `storage_arr` .